src/mon/MonitorDBStore.h

```diff
@@ -481,17 +458,35 @@ class MonitorDBStore
      *			    differ from the one passed on to the function)
      * @param last_key[out] Last key in the chunk
      */
-    void get_chunk_tx(TransactionRef tx, uint64_t max) override {
+    void get_chunk_tx(TransactionRef tx, uint64_t max_bytes,
+		      uint64_t max_keys) override {
       ceph_assert(done == false);
       ceph_assert(iter->valid() == true);
 
-      while (iter->valid()) {
+      while (iter->valid() && max_bytes > 0) {
 	string prefix(iter->raw_key().first);
 	string key(iter->raw_key().second);
 	if (sync_prefixes.count(prefix)) {
 	  bufferlist value = iter->value();
-	  if (!add_chunk_entry(tx, prefix, key, value, max))
+	  if (!tx->empty() &&
```

Comment:
i don't think this is correct. what if tx is empty at this moment?
shall we instead do something like
```c++
if (tx->get_bytes() + value.length() + key.size() +
    prefix.size() < max_bytes &&
    tx->get_keys() < max_keys) {
  // ...
}
```

as in http://pulpito.ceph.com/kchai-2019-11-15_17:09:33-rados-wip-kefu-testing-2019-11-13-1728-distro-basic-mira/, i found that lots of occurrences of "failed to reach quorum size 3 before timeout expired".

after checking the log, it turns out the mon which failed to join the quorum was keeping asking for its sync provider for the same paxos chunk.

see 
```
zgrep 'through version' /a/kchai-2019-11-15_17:09:33-rados-wip-kefu-testing-2019-11-13-1728-distro-basic-mira/4510653/remote/mira069/log/ceph-mon.a.log.gz | head -n100
```

and the provide kept sending empty mon sync replies
```
2019-11-15T17:38:31.670+0000 7fb2ed5cb700  1 -- [v2:172.21.6.110:3300/0,v1:172.21.6.110:6789/0] --> [v2:172.21.6.110:3301/0,v1:172.21.6.110:6790/0] -- mon_sync(chunk cookie 1006632961 lc 124 bl 26 bytes) v2 -- 0x557fc4f6b8c0 con 0x557fc37d3a80
```